src/lib/Libwin/passwd.c

```diff
@@ -3952,19 +3951,18 @@ setuser_with_password(char *user,
 	if ((user == NULL) || (decrypt_func == NULL))
 		return (0);
 
-	if(user_num == 0){
+	if(user_num == 0)
```

Comment:
Not sure why we still need user_num? It is a linked list, so it can be initialized by calling CLEAR_HEAD only once at server startup, and thus need not be called in every routine that you are calling here

Reply:
Okay, I'll change this.

Reply:
So, I've added a call to CLEAR_HEAD during server init (in pbsd_main.c), but also have one call to CLEAR_HEAD for pw_allpasswds in cache_usertoken_and_homedir since pbs_account has an entry point there and will crash if it's not initialised. This is called during the postinstall, which is why I've not removed that one.